lib/dotcom/trip_plan/location_nudger.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Dotcom.TripPlan.LocationNudger do
+  @moduledoc """
+  There are some locations that are fairly close to transit stops, but
+  for which all of the OpenStreetMap paths are not public. The
+  Massachusetts State House is a good example of this. The building
+  isn't publicly accessible according to OpenStreetMaps data, and
+  that's correct, because all of the entries into it are gated with
+  security guards. That said, plenty of people need to plan trips to
+  the State House.
+
+  This location nudger is responsible for nudging locations that
+  aren't publicly accessible, but that should still be possible to
+  plan trips to and from, to a sensible lat/long that can be used in
+  trip plans, e.g. the public side of a security entrance.
+  """
+
+  @state_house_address_string "24 Beacon St"
+  @state_house_zip_codes ["02108", "02133"]
+  @state_house_location %{latitude: 42.35861, longitude: -71.06297}
```

Comment:
Curious if you've considered a "dumber" solution that just defines before and after pairs that can be iterated through. The bonus would be easy to add new values which aren't state house specific.

```
@reconciliations [
  {"24 Beacon St, Boston, MA, 02108, USA", %{latitude: 42.35861, longitude: -71.06297}},
  {"24 Beacon St, Boston, MA, 02133, USA", %{latitude: 42.35861, longitude: -71.06297}},
  {"Massachusetts State House, 24 Beacon St, Boston, MA, 02108, USA", %{latitude: 42.35861, longitude: -71.06297}},
  {"Massachusetts State House, City of Boston, MA, USA", %{latitude: 42.35861, longitude: -71.06297}},
  {"State House News Service, 24 Beacon St, Boston, MA, 02133, USA", %{latitude: 42.35861, longitude: -71.06297}}
]
```

Reply:
I did consider that, and I'm not totally against it.

My reasoning was that I didn't want to have to list out each individual search query that would need to get nudged, and also if a new location crops up in search results like `State House Cafeteria, 24 Beacon St...`, then it wouldn't get nudged properly.

I'm honestly not sure what the best approach is. Matching on the zip may not be necessary. I did that a bit defensively to protect against a name like `Boston, MA Fake House, 24 Beacon St, Nowhere, MA, 02000`, which would get matched if we were just checking for `24 Beacon St` and `Boston, MA`. Admittedly that is a bit farfetched...

Reply:
Talked offline and decided to use geometry to figure out whether the coordinates are inside of the State House instead of basing it on the name.